plugins/modules/rabbitmq_vhost.py

```diff
@@ -97,46 +215,124 @@ def set_tracing(self):
+                ) % (self.name, response.status_code)
+                self.module.fail_json(msg=msg)
+        else:
+            return self._exec(["trace_on", "-p", self.name])
 
     def _disable_tracing(self):
-        return self._exec(['trace_off', '-p', self.name])
+        if self.login_host is not None:
+            response = self._request_vhost_api("put", data={"tracing": False})
+
+            if not response.ok:
+                msg = (
+                    "Error trying to disable tracing on vhost %s in rabbitmq. "
+                    "Status code '%s'."
+                ) % (self.name, response.status_code)
+                self.module.fail_json(msg=msg)
+        else:
+            return self._exec(["trace_off", "-p", self.name])
+
+    def _request_vhost_api(self, method, data=None):
+        try:
+            url = "%s://%s:%s/api/vhosts/%s" % (
+                self.login_protocol,
+                self.login_host,
+                self.login_port,
+                parse.quote(self.name, ""),
+            )
+            response = requests.request(
+                method=method,
+                url=url,
+                auth=(self.login_user, self.login_password),
+                verify=self.verify,
+                cert=(self.cert, self.key),
+                json=data,
+            )
+
+        except requests.exceptions.RequestException as exception:
+            msg = "Error in HTTP request (method %s) for user %s in rabbitmq." % (
+                method,
+                self.username,
+            )
+            self.module.fail_json(msg=msg, exception=exception)
+
+        return response
 
 
 def main():
     arg_spec = dict(
-        name=dict(required=True, aliases=['vhost']),
-        tracing=dict(default='off', aliases=['trace'], type='bool'),
-        state=dict(default='present', choices=['present', 'absent']),
-        node=dict(default='rabbit'),
+        name=dict(required=True, aliases=["vhost"]),
+        tracing=dict(default="off", aliases=["trace"], type="bool"),
+        state=dict(default="present", choices=["present", "absent"]),
+        node=dict(default="rabbit"),
+        login_user=dict(type="str", no_log=True, default="guest"),
+        login_password=dict(type="str", no_log=True, default="guest"),
+        login_host=dict(type="str"),
+        login_port=dict(type="str", default="15672"),
+        login_protocol=dict(type="str", default="http", choices=["http", "https"]),
+        ca_cert=dict(type="path", aliases=["cacert"]),
+        client_cert=dict(type="path", aliases=["cert"]),
+        client_key=dict(type="path", aliases=["key"]),
     )
 
-    module = AnsibleModule(
-        argument_spec=arg_spec,
-        supports_check_mode=True
-    )
+    module = AnsibleModule(argument_spec=arg_spec, supports_check_mode=True)
+
+    if not HAS_REQUESTS:
+        module.fail_json(
+            msg=missing_required_lib("requests"), exception=REQUESTS_IMP_ERR
+        )
```

Comment:
```suggestion
    if not HAS_REQUESTS:
        module.fail_json(
            msg=missing_required_lib("requests"), exception=REQUESTS_IMP_ERR
        )
```
So if users don't have it installed, their current playbooks will break?
If yes, it's a breaking change we should avoid.
Is it possible to avoid them? i.e. if those API specific options are not passed, the module will keep working as it does now?